packages/pwa/app/utils/url-config.js

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {getConfig} from './utils'
+import {urlPartPositions} from '../constants'
+import {rebuildPathWithParams} from './url'
+import {getAppOrigin} from 'pwa-kit-react-sdk/utils/url'
+import {getDefaultSite} from './site-utils'
+
+/**
+ * Rebuild the path with locale/site value to the path as url path or url query param
+ * based on url config
+ * @param {string} url - based url of the output url
+ * @param {object} configValues - object that contains values of url param config
+ * @return {string} - an output url
+ *
+ * @example
+ * //pwa-kit.config.json
+ * url {
+ *    locale: "query_param"
+ * }
+ * buildPathWithUrlConfig('/women/dresses', {locale: 'en-GB'})
+ *
+ * Returns
+ *  /women/dresses?locale=en-GB
+ *
+ *  @example
+ * //pwa-kit.config.json
+ * url {
+ *    locale: "path"
+ * }
+ * buildPathWithUrlConfig('/women/dresses', {locale: 'en-GB'})
+ *
+ * Returns
+ *  /en-GB/women/dresses
+ *
+ */
+export const buildPathWithUrlConfig = (url, configValues = {}) => {
+    const urlConfig = getUrlConfig()
+    if (!urlConfig || !Object.values(urlConfig).length) return url
+    if (!Object.values(configValues).length) return url
+    const queryParams = {}
+    let basePathSegments = []
+    const {defaultLocale} = configValues
+    const defaultSite = getDefaultSite()
+    // gather all default values for site and locale, id and aliases
+    const defaultValues = [defaultSite.id, defaultSite.alias, defaultLocale]
```

Comment:
We need to get both id and aliases for both locale and site so that we can check with the passing value inside `configValues`. For now, we don't have alias for locale yet.